relay-server/src/actors/global_config.rs

```diff
@@ -256,17 +266,28 @@ impl GlobalConfigService {
         match result {
             Ok(Ok(config)) => {
                 let mut success = false;
+                // No global status implies an outdated upstream, in that case, we fall back on
+                // the old behaviour of using the global config that we receive even if it might
+                // not be the valid one.
+                let is_ready = config
+                    .global_status
+                    .map(|stat| stat.is_ready())
```

Comment:
This could be a `map_or(true, |stat| stat.is_ready()))`

Reply:
nice tip! dealing with options is really an art in rust lol